pkg/trace/agent/agent.go

```diff
@@ -339,9 +339,10 @@ func (a *Agent) Process(p *api.Payload) {
 var _ api.StatsProcessor = (*Agent)(nil)
 
 func (a *Agent) processStats(in pb.ClientStatsPayload, lang, tracerVersion string) pb.ClientStatsPayload {
-	if features.Has("disable_cid_stats") || a.conf.FargateOrchestrator == fargate.Unknown {
-		// this functionality is disabled by the disable_cid_stats feature flag
-		// or we're not in a Fargate instance.
+	enableContainers := features.Has("enable_cid_stats") || (a.conf.FargateOrchestrator != fargate.Unknown)
+	if !enableContainers || features.Has("disable_cid_stats") {
```

Comment:
Maybe you can write this differently or on one line? I have been looking at it for over 5 minutes now (no kidding) and can't figure out how it works. 

P.S. Might be just me! Very high chance 🙂